agent/internal/container/container_api_test.go

```diff
@@ -215,3 +218,134 @@ func TestContainer(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerStatus(t *testing.T) {
+	log.SetLevel(log.TraceLevel)
+
+	t.Log("building client")
+	rawCl, err := dclient.NewClientWithOpts(dclient.WithAPIVersionNegotiation(), dclient.FromEnv)
+	require.NoError(t, err)
+	defer func() {
+		if cErr := rawCl.Close(); cErr != nil {
+			t.Logf("closing docker client: %s", cErr)
+		}
+	}()
+	cl := docker.NewClient(rawCl)
+
+	tests := []struct {
+		name                string
+		image               string
+		entrypoint          []string
+		signalAtState       cproto.State
```

Comment:
```signalAtState``` is unused I think

Reply:
Yep! I had originally been also checking for when the container had entered the `Starting` status, but it ended up being unnecessary since the main focus was on killing it off once the docker container was created.

I'll get that removed